crates/red_knot/src/args.rs

```diff
@@ -299,3 +306,77 @@ pub(crate) enum TerminalColor {
     /// Never display colors.
     Never,
 }
+/// Either a path to a TOML configuration file (`knot.toml`),
+/// or a TOML `<KEY> = <VALUE>` pair
+/// (such as you might find in a `knot.toml` configuration file)
+/// overriding a specific configuration option.
+/// Later values override earlier ones, and values passed via this `--config` argument
+/// take precedence over values in other configuration files.
```

Comment:
`--config` should only accept configuration arguments. We'll use `--config-file` to allow overriding a configuration file.